includes/data-stores/class-wcs-related-order-store-cpt.php

```diff
@@ -40,11 +40,14 @@ public function __construct() {
 	 * @return array
 	 */
 	public function get_related_order_ids( WC_Order $subscription, $relation_type ) {
+		// Filter out the new 'wc-checkout-draft' status, as it is not a valid order status for our purposes.
+		$statuses = array_keys( wc_get_order_statuses() );
+		unset( $statuses[ array_search( 'wc-checkout-draft', $statuses, true ) ] );
```

Comment:
I'm trying to think if this will cause issues. Excluding `wc-checkout-draft` orders from a subscription's related order cache appears at first glance to me to be too heavy handed. 

This function is used to populate the cache and I'd be concerned that if the order status is updated to another (valid) status then we'd need to make sure it's updated then to include that order. ie if a switch order changed from `wc-checkout-draft` to `processing` the cache would need to be updated to include that order, as it is now a valid status. 

For some background, the cache is currently populated/updated when an order is given a [certain relation](https://github.com/Automattic/woocommerce-subscriptions-core/blob/7.2.0/includes/data-stores/class-wcs-related-order-store-cached-cpt.php#L166-L168). For example if an order is given the `'switch'` relationship, meta is added to the order to indicate it's that type of order and the subscription's related order cache is updated by adding the new ID to the start of the cache {[code ref](https://github.com/Automattic/woocommerce-subscriptions-core/blob/7.2.0/includes/data-stores/class-wcs-related-order-store-cached-cpt.php#L236-L237)}. Its what we called a _persistant cache_, its actively updated rather than invalidated and repopulated. 

So given `wc-checkout-draft` orders are excluded, its my understanding we'd need to add and remove items from the order cache, when it's status is updated. That complicates things a lot.